docs/reference/api_examples.yml

```diff
@@ -249,6 +249,9 @@
 
   doc: |
     Get a list of all datasets in a deployment.
+    This includes manifest (attached) and non-manifest (unattached) datasets.
+    A dataset is non-manifest when it is moving between nodes.
+    The ``primary`` and ``path`` values of non-manifest datasets will be ``null``.
```

Comment:
We don't include `maximum_size` in the response at all if it's not set on the dataset.  It would make sense to be consistent with that rather than have some behaviors where clients have to check for presence and others where they have to check for None.